documentation/architecture.md

```diff
@@ -0,0 +1,247 @@
+# Clear Containers Architecture
```

Comment:
+1 - this is superb. Just some minor nits.

We do need to either update or just remove the following which is (now) 2.0-specific (missing all the new 2.1 goodness ;-):

- https://github.com/01org/cc-oci-runtime/blob/master/DESIGN.rst#code-flow-with-docker-1-12